src/api/worker/rest/EntityRestCache.ts

```diff
@@ -255,110 +243,186 @@ export class EntityRestCache implements IEntityRestCache {
 		return entitiesFromServer.concat(entitiesInCache)
 	}
 
-	private async _loadRange<T extends ListElementEntity>(typeRef: TypeRef<T>, listId: Id, start: Id, count: number, reverse: boolean): Promise<T[]> {
+	async loadRange<T extends ListElementEntity>(typeRef: TypeRef<T>, listId: Id, start: Id, count: number, reverse: boolean): Promise<T[]> {
+		const typeModel = await resolveTypeReference(typeRef)
+		if (
+			typeRef.app === "monitor"
+			|| this.ignoredTypes.some(ref => isSameTypeRef(typeRef, ref))
+			|| typeModel.values._id.type !== ValueType.GeneratedId // we currently only store ranges for generated ids
+		) {
+			return this.entityRestClient.loadRange(typeRef, listId, start, count, reverse)
+		}
 
 		const range = await this.storage.getRangeForList(typeRef, listId)
 
-		if (range == null) {
-
-			// Create a new range and load everything
-			const entities = await this.entityRestClient.loadRange(typeRef, listId, start, count, reverse)
-
-			// Initialize a new range for this list
-			await this.storage.setNewRangeForList(typeRef, listId, start, start)
-
-			// The range bounds will be updated in here
-			return this.handleElementRangeResult(typeRef, listId, start, count, reverse, entities, count)
+		if (
+			range == null
+		) {
+			await this.createNewRange(typeRef, listId, start, count, reverse)
+		} else if (
+			!firstBiggerThanSecond(start, range.upper) && !firstBiggerThanSecond(range.lower, start)
```

Comment:
Took me a minute to understand, maybe some comment might make it easier (like "start is inside range")